enterprise/admin/views.py

```diff
@@ -367,16 +371,29 @@ def get_enterprise_customer_user_queryset(self, request, search_keyword, custome
             page_size (int): Number of learners displayed in each paginated set.
         """
         page = request.GET.get('page', 1)
-        learners = EnterpriseCustomerUser.objects.filter(enterprise_customer__uuid=customer_uuid)
-        user_ids = learners.values_list('user_id', flat=True)
-        matching_users = User.objects.filter(pk__in=user_ids)
-        if search_keyword is not None:
-            matching_users = matching_users.filter(
-                Q(email__icontains=search_keyword) | Q(username__icontains=search_keyword)
+        if search_keyword:
+            enterprise_learners = self._get_searched_enterprise_customer_users(
+                search_keyword,
+                customer_uuid,
             )
-        matching_user_ids = matching_users.values_list('pk', flat=True)
-        learners = learners.filter(user_id__in=matching_user_ids)
-        return paginated_list(learners, page, page_size)
+        else:
+            enterprise_learners = EnterpriseCustomerUser.objects.filter(
+                enterprise_customer__uuid=customer_uuid,
+            )[:DJANGO_ADMIN_MANAGE_LEARNERS_LIMIT]
+        return paginated_list(enterprise_learners, page, page_size)
+
+    def _get_searched_enterprise_customer_users(self, search_keyword, customer_uuid):
+        """
+        Helper to return a paginated list of enterprise customer users
+        who match some search criteria based on corresponding ``core.User`` records.
```

Comment:
Nit: maybe use a different term than "paginated", e.g. "truncated".

Reply:
Good point, I originally used "paginated" based on returning the result of the `paginated_list()` function, but I'm no longer using that function here in this new helper function.